services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/events.py

```diff
@@ -1,5 +1,235 @@
+            dynamic_sidecar_settings=dynamic_sidecar_settings,
+            dynamic_sidecar_network_id=dynamic_sidecar_network_id,
+            swarm_network_id=swarm_network_id,
+            settings=settings,
+        )
+        logger.debug(
+            "dynamic-sidecar create_service_params %s",
+            pformat(dynamic_sidecar_create_service_params),
+        )
+
+        dynamic_sidecar_id = await create_service_and_get_id(
+            dynamic_sidecar_create_service_params
+        )
+
+        dynamic_sidecar_node_id = await get_node_id_from_task_for_service(
+            dynamic_sidecar_id, dynamic_sidecar_settings
+        )
+
+        dynamic_sidecar_proxy_create_service_params = await get_dynamic_proxy_spec(
+            scheduler_data=scheduler_data,
+            dynamic_sidecar_settings=dynamic_sidecar_settings,
+            dynamic_sidecar_network_id=dynamic_sidecar_network_id,
+            swarm_network_id=swarm_network_id,
+            swarm_network_name=swarm_network_name,
+            dynamic_sidecar_node_id=dynamic_sidecar_node_id,
+        )
+        logger.debug(
+            "dynamic-sidecar-proxy create_service_params %s",
+            pformat(dynamic_sidecar_proxy_create_service_params),
+        )
+
+        # no need for the id any longer
+        await create_service_and_get_id(dynamic_sidecar_proxy_create_service_params)
+
+        # update service_port and assing it to the status
+        # needed by CreateUserServices action
+        scheduler_data.service_port = extract_service_port_from_compose_start_spec(
+            dynamic_sidecar_create_service_params
+        )
+
+        # finally mark services created
+        scheduler_data.dynamic_sidecar.were_services_created = True
+
+
+class GetStatus(DynamicSchedulerEvent):
+    """
+    Inspects the dynamic-sidecar, and store some information about the contaiers.
+    """
+
+    @classmethod
+    async def will_trigger(cls, app: FastAPI, scheduler_data: SchedulerData) -> bool:
+        return (
+            scheduler_data.dynamic_sidecar.status.current == DynamicSidecarStatus.OK
+            and scheduler_data.dynamic_sidecar.is_available == True
+        )
+
+    @classmethod
+    async def action(cls, app: FastAPI, scheduler_data: SchedulerData) -> None:
+        dynamic_sidecar_client = get_dynamic_sidecar_client(app)
+        dynamic_sidecar_endpoint = scheduler_data.dynamic_sidecar.endpoint
+
+        containers_inspect: Optional[
+            Dict[str, Any]
+        ] = await dynamic_sidecar_client.containers_inspect(dynamic_sidecar_endpoint)
+        if containers_inspect is None:
+            # After the service creation it takes a bit of time for the container to start
+            # If the same message appears in the log multiple times in a row (for the same
+            # service) something might be wrong with the service.
+            logger.warning(
+                "No container present for %s. Usually not an issue.",
```

Comment:
I am not sure that warning is helpful in any ways... you said it takes 50 secs to start the sidecar, does that mean we will see this message 10 times for each services?

Reply:
`GetStatus.action()`  will starts running after `CreateSidecars.action()` runs. Sometimes the containers are not present at that time, on the next iteration the containers are already there.
I wanted a warning message here to help with debugging (especially if something happens in production/staging)
Any suggestions for a better message are welcomed.